src/main/java/frc/robot/scoreStates/AmpState.java

```diff
@@ -8,49 +8,80 @@
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.Commands;
 import frc.robot.Constants;
+import frc.robot.commandGroups.CommandGroupsConstants;
+import frc.robot.subsystems.elevator.Elevator;
+import frc.robot.subsystems.gripper.Gripper;
+import frc.robot.subsystems.gripper.GripperConstants;
 import frc.robot.swerve.SwerveDrive;
 import java.util.Set;
 import lib.Utils;
 
 public class AmpState implements ScoreState {
```

Comment:
Think about a situation where _Andromeda_ is against the Amp's wall there is not enough space to rotate. You could choose to rotate the arm instead, although the distance is very short.

Reply:
This is a very extreme case that would not happen because the driver is not stupid.

Reply:
And I'm not saying they are. But think of a situation when the robot is pinned by by an opponent against the Amp, then the driver executed autonomous scoring immediately after the pin ends. The software should take off from the driver as much cognitive load as possible.

Reply:
Again the driver is not stupid he just will not do it against the wall

Reply:
It has nothing with the driver being "stupid". It's about making their lives as easy as possible and the software should be versatile and able to handle rare cases, too (and it's just a few more lines of code). Another reason to do this is just to push our team's (software team) boundaries and make the best software we can, because why settle for less?

But, if you are still not convinced, you can resolve this thread.